ElectronClient/gui/NoteEditor/NoteBody/CodeMirror/CodeMirror.tsx

```diff
@@ -370,14 +370,24 @@ function CodeMirror(props: NoteBodyEditorProps, ref: any) {
 	const editorReadOnly = props.visiblePanes.indexOf('editor') < 0;
 
 	function renderEditor() {
+		// The editor needs to be kept up to date on the actual space that it's filling
+		// Ogherwise we can get some rendering errors when the editor size is changed
+		// (this can happen when switching layout of when toggling sidebars for example)
+		const editorStyle = Object.assign({ width: 0, height: 0 }, styles.editor);
+		if (rootRef.current && props.visiblePanes.includes('editor')) {
+			const rootSize = rootRef.current.getBoundingClientRect();
+			editorStyle.width = !props.visiblePanes.includes('viewer') ? rootSize.width : Math.floor(rootSize.width / 2);
+			editorStyle.height = rootSize.height;
+		}
```

Comment:
Could you use `useMemo` to generate the `editorStyle` property please? Otherwise the `Editor` component will re-render every time even if the style doesn't really change (Object.assign create a new object every time).

Reply:
Good idea, I copied the style you had in AceEditor for keeping track of root size. This also makes the code a bit cleaner as well.